src/psyclone/tests/psyir/transformations/intrinsics/dotproduct2code_trans_test.py

```diff
@@ -328,44 +316,52 @@ def test_apply_calls_validate():
             "'NoneType'." in str(info.value))
 
 
-@pytest.mark.parametrize("dim1,dim2", [("10", "10"), (":", "10"), ("10", ":")])
-def test_apply_known_dims(tmpdir, fortran_reader, fortran_writer, dim1, dim2):
+@pytest.mark.parametrize("dim1,dim2,upper_bound",
+                         [("10", "10", "10"), (":", "10", "10"),
+                          ("10", ":", "10"), ("10", "n", "10"),
+                          ("n", "10", "n"), (":", "n", "n"),
+                          ("n", ":", "n")])
+def test_apply_known_dims(
+        tmpdir, fortran_reader, fortran_writer, dim1, dim2, upper_bound):
     '''Test that the DotProduct2CodeTrans apply method produces the
     expected PSyIR when at least one of the vectors has a known
-    dimension.
+    dimension. Also check that the correct precision is used.
 
     '''
     code = (
-        f"subroutine dot_product_test(v1,v2)\n"
-        f"real,intent(in) :: v1({dim1}), v2({dim2})\n"
-        f"real :: result\n"
+        f"subroutine dot_product_test(v1,v2,n)\n"
+        f"integer,intent(in) :: n\n"
+        f"real*4,intent(in) :: v1({dim1}), v2({dim2})\n"
+        f"real*4 :: result\n"
         f"result = dot_product(v1,v2)\n"
         f"end subroutine\n")
     expected = (
-        "  integer :: i\n"
-        "  real :: res_dot_product\n\n"
-        "  res_dot_product = 0.0\n"
-        "  do i = 1, 10, 1\n"
-        "    res_dot_product = res_dot_product + v1(i) * v2(i)\n"
-        "  enddo\n"
-        "  result = res_dot_product\n\n")
+        f"  integer :: i\n"
+        f"  real*4 :: res_dot_product\n\n"
+        f"  res_dot_product = 0.0\n"
+        f"  do i = 1, {upper_bound}, 1\n"
+        f"    res_dot_product = res_dot_product + v1(i) * v2(i)\n"
+        f"  enddo\n"
+        f"  result = res_dot_product\n\n")
     check_trans(code, expected, fortran_reader, fortran_writer, tmpdir)
 
 
 def test_apply_unknown_dims(tmpdir, fortran_reader, fortran_writer):
     '''Test that the DotProduct2CodeTrans apply method produces the
-    expected PSyIR when neither of the vectors have a known size.
+    expected PSyIR when neither of the vectors have a known size. Also
+    check that the correct precision is used.
 
```

Comment:
Please add a new test that the validate() rejects a case where the two DOT_PRODUCT arguments have different precision.

Reply:
I did not add a check for this in the validation so there is no associated test. I did start but then decided that this is really a job for the node itself, not the transformation. Similarly for mixed datatypes. My argument is that if the code is already written in this form in the PSyIR then it should already have been checked. I could add a reference to the appropriate issue in the validate method and/or test, so it is not forgotten?

Reply:
Actually, I looked at the Fortran standard and it says nothing about precision. Anyway, if you're not checking for it please can you update the docstring :-)

Reply:
Updated the docstrings in the transformation and transformation test.